flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/codegen/LongHashJoinGenerator.scala

```diff
@@ -44,11 +44,12 @@ object LongHashJoinGenerator {
         joinType == HashJoinType.ANTI ||
         joinType == HashJoinType.PROBE_OUTER) &&
         filterNulls.forall(b => b) &&
-        keyType.getFieldTypes.length == 1 && {
-      val t = keyType.getTypeAt(0)
-      t == InternalTypes.LONG || t == InternalTypes.INT || t == InternalTypes.SHORT ||
-          t == InternalTypes.BYTE || t == InternalTypes.FLOAT || t == InternalTypes.DOUBLE ||
-          t.isInstanceOf[DateType] || t.isInstanceOf[TimestampType] || t == InternalTypes.TIME
+        keyType.getFieldCount == 1 && {
+      keyType.getTypeAt(0) match {
+        case _: BigIntType | _: IntType | _: SmallIntType | _: TinyIntType |
```

Comment:
Use type families instead?

Reply:
Decimal is not supported here, so type families cannot be used. (Just because we haven't supported it yet)
I think I can use type families in `TypeCheckUtils`.